packages/flutter_tools/test/general.shard/runner/flutter_command_test.dart

```diff
@@ -287,6 +287,45 @@ void main() {
         SystemClock: () => clock,
         Usage: () => usage,
       });
+
+      testUsingContext('command release lock on kill signal', () async {
+        mockTimes = <int>[1000, 2000];
+        final Completer<void> completer = Completer<void>();
+        setExitFunctionForTests((int exitCode) {
+          expect(exitCode, 0);
+          restoreExitFunction();
+          completer.complete();
+        });
+        final DummyFlutterCommand flutterCommand = DummyFlutterCommand(
+            commandFunction: () async {
+              await Cache.lock();
+              final Completer<void> c = Completer<void>();
+              await c.future;
+              return null; // unreachable
+            }
+        );
+        unawaited(flutterCommand.run());
+        await Future<void>.delayed(const Duration(milliseconds: 50));
```

Comment:
Does this need to be a delay, or can you complete a Completer at the top of the `commandFunction` closure, and await its Future here?

Reply:
I am not quite sure that I understood the comment, sorry :(
The delay is added to make sure that the lock is acquired because without it the test is failing because the lock will not be acquired by the time of checking it (hence `StateError` will be thrown)
Probably I am missing or misunderstanding something.
Could you please give a bit more details?

Reply:
Sure, like:
```dart
final Completer<void> cacheLockedCompleter = Completer<void>();
final DummyFlutterCommand flutterCommand = DummyFlutterCommand(
  commandFunction: () async {
    await Cache.lock();
    cacheLockedCompleter.complete();
    final Completer<void> c = Completer<void>();
    await c.future;
    return null; // unreachable
  },
);
unawaited(flutterCommand.run());
await cacheLockedCompleter.future;
...
```

Reply:
Got it, thank you very much